seumich/views.py

```diff
@@ -112,12 +112,13 @@ def get_queryset(self):
         student_list = []
         if self.query_user:
             # Filtering for id >= 0 eliminates "Bad Value"-type results.
-            query_user_list = self.query_user.split(' ')
-            q_list = [Q(username__icontains=x) for x in query_user_list]
-            q_list += [Q(univ_id__icontains=x) for x in query_user_list]
-            q_list += [Q(first_name__icontains=x) for x in query_user_list]
-            q_list += [Q(last_name__icontains=x) for x in query_user_list]
+            q_list = [Q(username__icontains=self.query_user)]
+            q_list += [Q(univ_id__icontains=self.query_user)]
+            q_list += [Q(first_name__icontains=self.query_user)]
+            q_list += [Q(last_name__icontains=self.query_user)]
+            q_list += [Q(full_name__icontains=self.query_user)]
             student_list = (Student.objects.filter(id__gte=0)
+                            .annotate(full_name=(Concat('first_name', V(' '), 'last_name')))
```

Comment:
not sure why we need to use V(' ')? Can we use ' ' directly?

Reply:
What this is doing is actually pretty cool. This is part of the django ORM the values in quotes are actual references to columns in the model. It has to use V (which is a shortcut for django.db.models.Value) so that it doesn't reference the database column. If it just used ' ' it would be looking for a database column in the model that is a blank space. 

This syntax is creating a "virtual" column in the database, it's just in the SQL syntax and in SQL it would appear like SELECT * from <table> where first_name + ' ' + last_name = '<something>' 

https://docs.djangoproject.com/en/1.11/ref/models/database-functions/#django.db.models.functions.Concat